website/docs/guides/markdown-features/markdown-features-intro.mdx

```diff
@@ -230,8 +230,8 @@ Markdown can embed HTML elements, and [`details`](https://developer.mozilla.org/
 
 :::info
 
-- You should not break lines between the starting or ending tag of the `<summary>` element and its content or an extra `<p>` element is inserted.
-- Blank lines between the `<summary>` element and the first paragraph of the detailed content are optional in Docusaurus but strongly recommended to ensure MDX portability with other site generators.
-- Blank lines between the nested `<details>` element and the paragraphs just around it are optional.
+- You may want to keep your `<summary>` on a single line. [MDX creates extra HTML `<p>` paragraphs for line breaks.](https://mdxjs.com/migrating/v2/#jsx)
+- You should add a blank line between the `<summary>` element and the first paragraph of the detailed content, or React will emit a warning in the console in the DevTools: `Warning: validateDOMNesting(...): <summary> cannot appear as a descendant of <p>.`
+- You can optionally add a blank line between the nested `<details>` element and the paragraphs just around the nested `<details>` to clarify the semantic boundary between the two.
```

Comment:
I'm going to remove that too because this is incorrect

A blank line after summary is not required. In general, the only rule that apply is how MDX creates extra paragraphs on line breaks

These 2 forms do not involve blank lines and are perfectly valid:

```jsx
<details>
  <summary>hey</summary>
  <>hoo</>
</details>
```

```jsx
<details><summary>hey</summary>hoo</details>
```

Reply:
> on line breaks

If the paragraph candidate is composed of only inline JSXes (`<...>...</...>`) and whitespaces (including line breaking), it's not enclosed by a `<p>`.

```mdx
<>foo</>bar<>baz</>
```

↑ will be enclosed by `<p>` because `bar` is bare

```
<>foo</><>bar
baz</><>qux</>
<>quux</>
```

↑ won't be enclosed by `<p>` because they consist of 3 inline JSXes and a newline (between `<>qux</>` and `<>quux</>`)


```mdx
<details><summary>hey</summary>hoo</details>
```

↑The entire line is the paragraph candidate, but all the content is enclosed by the *inline* JSX, so no paragraph is inserted.

```mdx
<details>
  <summary>hey</summary>
  <>hoo</>
</details>
```

↑Its paragraph candidate is (block JSX narrows the range of paragraph candidate):

```mdx
  <summary>hey</summary>
  <>hoo</>
```

The *entire* content of both 2 *lines* is *enclosed* by an inline JSX tag (`<summary>` and `<>`).
With a notation like `<>hoo</>`, you can't add another paragraph or a code block there.


```mdx
<summary>hey</summary>
<>foo
bar
baz</>
<>qux
quux</>
```

This is also OK because no bare lines (not enclosed by inline JSX)

```mdx
<summary>hey</summary>
<>foo
bar</>
baz
<>qux
quux</>
```

`baz` is bare there, so a paragraph will be inserted.

I feel it's difficult for me to explain it to beginners simply.